specs/ReleaseChannelPreferenceSet.md

```diff
@@ -0,0 +1,325 @@
+These options can be used together to specify which channels the WebView2 loader should search for
+in order of least to most stable: Canary -> Dev -> Beta -> Stable WebView2 Runtime.
+For example, you can set `ShouldReverseChannelSearchOrder` to true and specify {Beta, Stable}
+to tell the WebView2 loader to try running the app with Beta channel if it exists on the machine,
+and otherwise fall back on the stable WebView2 Runtime. With the new options you can:
+- Run automated tests against different channels on a machine with multiple channels installed
+- Target a particular channel when flighting
+- Apply different channels to different flighting rings
+- Fall back on a more stable preview channel
+
+# Examples
+## RuntimeChannelPreferenceSet
+The runtime channel preference set and channel search order are specified on the
+CoreWebView2EnvironmentOptions prior to creating the WebView2 environment.
+```c#
+CoreWebView2Environment _webViewEnvironment;
+async System.Threading.Tasks.Task CreateEnvironmentAsync() {
+    CoreWebView2RuntimeChannel channels =
+        CoreWebView2RuntimeChannel.Beta | CoreWebView2RuntimeChannel.Stable;
+    CoreWebView2EnvironmentOptions customOptions = new CoreWebView2EnvironmentOptions
+    (
+        shouldReverseChannelSearchOrder: true,
+        runtimeChannelPreferenceSet: channels
+    );
+    // Use GetAvailableCoreWebView2BrowserVersionStringWithOptions to check which
+    // channel is used with the custom options. If Beta channel was not found,
+    // install it on the device.
+    string version =
+        CoreWebView2Environment.GetAvailableBrowserVersionStringWithOptions(
+            browserExecutableFolder: null,
+            environmentOptions: customOptions);
+    if (!IsBetaChannel(version))
+    {
+        InstallChannel(CoreWebView2RuntimeChannel.Beta);
+    }
+    // If the loader is unable to find a valid installation from the runtime
+    // channel preference set, environment creation will fail.
+    _webViewEnvironment = await CoreWebView2Environment.CreateAsync(
+        options: customOptions);
+    if (_webViewEnvironment != null)
+    {
+        CheckChannel(_webviewEnvironment.BrowserVersionString());
+    }
+}
+```
+```cpp
+void AppWindow::InitializeWebViewEnvironment()
+{
+    auto options = Microsoft::WRL::Make<CoreWebView2EnvironmentOptions>();
+    CHECK_FAILURE(options->put_ShouldReverseChannelSearchOrder(TRUE));
+    COREWEBVIEW2_RUNTIME_CHANNEL channels =
+        COREWEBVIEW2_RUNTIME_CHANNEL_BETA |
+        COREWEBVIEW2_RUNTIME_CHANNEL_STABLE;
+    CHECK_FAILURE(options->SetRuntimeChannelPreferenceSet(channels));
+    // Use GetAvailableCoreWebView2BrowserVersionStringWithOptions to check which
+    // channel is used with the custom options. If Beta channel was not found,
+    // install it on the device.
+    wil::unique_cotaskmem_string version;
+    HRESULT hr = GetAvailableCoreWebView2BrowserVersionStringWithOptions(
```

Comment:
```suggestion
    HRESULT hr = CoreWebView2Environment::GetAvailableCoreWebView2BrowserVersionString(
```

Reply:
No change required. This is our COM flat global function.